src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,81 @@
+package duke;
+
+import java.io.IOException;
+
+import duke.exceptions.DukeException;
+import duke.tasks.TaskList;
+
+/**
+ * Represents a Duke object that runs the duke program.
+ */
+public class Duke {
+    private TaskList tasks;
+    private Ui ui;
+    private Storage storage;
+
+    /**
+     * Constructor for Duke.
+     * Creates a new Duke object with the given file path.
+     * Loads the tasks from the file path.
+     * If the file path does not exist, creates a new file.
+     *
+     * @param filePath The path to the file where the tasks are stored.
+     */
+    public Duke(String filePath) {
+        ui = new Ui();
+        try {
+            storage = new Storage(filePath);
+            tasks = new TaskList(storage.load());
+        } catch (IOException e) {
```

Comment:
Good Error catching here and creating a new empty list rather than terminating the program